pkg/kubectl/scale.go

```diff
@@ -180,11 +197,63 @@ func (scaler *ReplicationControllerScaler) Scale(namespace, name string, newSize
 	return nil
 }
 
-// ScalerClient abstracts access to ReplicationControllers.
+// ScaleSimple is responsible for updating job's parallelism.
+func (scaler *JobScaler) ScaleSimple(namespace, name string, preconditions *ScalePrecondition, newSize uint) (string, error) {
+	job, err := scaler.c.GetJob(namespace, name)
+	if err != nil {
+		return "", ControllerScaleError{ControllerScaleGetFailure, "Unknown", err}
+	}
+	if preconditions != nil {
+		if err := preconditions.ValidateJob(job); err != nil {
+			return "", err
+		}
+	}
+	parallelism := int(newSize)
+	job.Spec.Parallelism = &parallelism
+	if _, err := scaler.c.UpdateJob(namespace, job); err != nil {
+		if errors.IsInvalid(err) {
+			return "", ControllerScaleError{ControllerScaleUpdateInvalidFailure, job.ResourceVersion, err}
+		}
+		return "", ControllerScaleError{ControllerScaleUpdateFailure, job.ResourceVersion, err}
+
+	}
+	return "scaled", nil
+}
+
+// Scale updates a Job to a new size, with optional precondition check (if preconditions is not nil),
+// optional retries (if retry is not nil), and then optionally waits for parallelism to reach desired
+// number, which can be less than requested based on job's current progress.
+func (scaler *JobScaler) Scale(namespace, name string, newSize uint, preconditions *ScalePrecondition, retry, waitForReplicas *RetryParams) error {
+	if preconditions == nil {
+		preconditions = &ScalePrecondition{-1, ""}
+	}
+	if retry == nil {
+		// Make it try only once, immediately
+		retry = &RetryParams{Interval: time.Millisecond, Timeout: time.Millisecond}
+	}
+	cond := ScaleCondition(scaler, preconditions, namespace, name, newSize)
+	if err := wait.Poll(retry.Interval, retry.Timeout, cond); err != nil {
+		return err
+	}
+	if waitForReplicas != nil {
+		job, err := scaler.c.GetJob(namespace, name)
+		if err != nil {
+			return err
+		}
+		return wait.Poll(waitForReplicas.Interval, waitForReplicas.Timeout,
+			scaler.c.JobHasDesiredParallelism(job))
+	}
+	return nil
+}
+
+// ScalerClient abstracts access to ReplicationControllers & Jobs.
```

Comment:
It feels like this interface isn't providing a valuable abstraction.  What practical value does it have if we can't abstract out how different things need to scale.

Perhaps passing through a `client.Interface` and letting each scaler choose how to make use it of would be more useful.